build.sh

```diff
@@ -4,21 +4,20 @@ set -xeuo pipefail
 
 basedir=$(dirname "$(readlink -f "$0")")
 
-# Make sure conda is updated to a version that supports the --no-capture-output option
-conda install -y -n base -c conda-forge "conda>=4.13.0"
+export MAMBA_ROOT_PREFIX=$(conda info --base)
 
-# Create conda env containing only conda-lock
-conda create -y -n gedi_subset conda-lock
+# Utilizing micromamba
+mamba install -y -n base -c conda-forge "micromamba>=1.5.0"
```

Comment:
Can we stick with `conda` here so that this script can still work in the ADE without requiring us to install `mamba` in our ADE workspaces? If `mamba` is now definitely part of _all_ workspace images, then you can ignore this comment. (`mamba` was missing from my workspace, but I created my workspace a few months ago, and I didn't create a new workspace to see if this is the case.)

At a mimimum, keep `conda-lock` on this line, as we still need it for updating our lock file when we want/need to make dependency changes.

Reply:
Will have to check on `mamba` but it sounds like it should be part of all workspace images. We got by registering the algorithm Friday by changing the outdated image to the vanilla:3.0.1. We should probably still take a look at updating the build process to use mamba or libmamba-solver. Maybe next Thursday? I'll move this to draft as it's not pressing at the moment

Reply:
`mamba` is in all the workspaces now, however an upcoming change will shift to ensuring `libmamba` solver is installed. I agree we can leave them as conda commands and try to rely on libmamba solver option.

Reply:
Using `micromamba` does the trick for the build on the subsequent lines where the lock file is _not_ used, but using `mamba` on this particular line makes little to no performance difference since this line installs _only_ `micromamba` (and `conda-lock` when you add that back), so the dependency resolution effort is "trivial" in comparison to what is required for the dependencies required by our code.

The lines in the build that now use `micromamba` where no lock file is used are the significant lines (except for this line) requiring the faster resolver since they produce very large dependency trees, so that's a good addition there.

Therefore, whether we use `conda` or `mamba` on this line has little to no impact on the build time, so the decision on which to use is perhaps only driven by whether or not `mamba` is now a standard install on all workspace images (now and into the future).

My original motivation for this script was to make it such that it works not only on the DPS images, but also on the ADE images (and even on our own laptops), so that we can be fairly confident that a successful execution of this script in an ADE workspace (or on our laptops) will result in a successful execution during DPS registration, which then indicates that we should be very unlikely to run into a dependency issue during algorithm execution.

This was driven by the extremely frustrating experience we had early on where it would work without issue in the ADE, but fail during DPS job execution (but _not_ during registration), which led to the introduction of `conda lock`, along with the small bit of code at the end of this script, which would cause DPS registration to fail fast if the code failed, avoiding a much longer wait time to see things fail during algorithm execution rather than during algorithm registration.

If we end up sticking with `mamba` on this line (if the intent that all existing and future images for both ADE and DPS will come with `mamba` preinstalled), then that's fine with me, but if so, then you'll also need to update this paragraph in the `README.md` file as well:

> If you plan to do any development work outside of the ADE (such as on your local
workstation), perform the steps above in that location as well.  **NOTE:** _This
means that you must have `conda` installed (see [conda installation]) in your
desired development location outside of the ADE workspace._

along with updating/changing this link ref for `[conda installation]` near the bottom of `README.md`.

I'm happy to discuss further, as you suggest, whenever is convenient.

Reply:
We should probably close this PR. I don't think we're going to make updates to the process in this way.
1. We should rely on `conda` it's more standard
2. We should leverage `libmamba` solver when available (which on MAAP it will be, but elsewhere may not be)
3. We should document that for external potential users.

Reply:
According to https://github.com/conda/conda/releases/tag/23.9.0, `libmamba` will be the default solver for `conda` 23.10.0 onwards. Could enable it today using `conda config --set solver libmamba`.